### DISCUSSION TITLE
oxclint/format rule: import policy

### DISCUSSION THREAD

**Discussion by Goldziher** [oxc-project/oxc#5719] [Feature Request]
Hi, i'd like to suggest an autofix rule for import sorting enhancement: Add an ability to set import preferenance. For example, I prefer to restructure named imports from global namespaces, or some people prefer explicitly importing global namespaces.

### Global Namespace
The React global namespace is used for typing

```typescript
import React // not required in recent versions, is available as UMD global etc.

const MyComponent: React.FC = () => {
 // something
}
```

### Explicit Imports
The types are imported from react explicitly. 

```typescript
import { type FC } from 'react' 

const MyComponent: FC = () => {
 // something
}
```

Or the opposite:

### Explicit Imports
Vitest functions are imported directly.

```typescript
import { describe} from 'vitest'

describe("Some Test Suite", () => {
   // ...
}) 
```

### Global Namespace

The vitest globals are configured to be globally available.

```typescript
describe("Some Test Suite", () => {
   // ...
}) 
```

**Comment by Boshen** [comment]
https://github.com/oxc-project/oxc/issues/14253
